### PR TITLE
Add a Servlet to save contact form data into Datastore. 

### PR DESCRIPTION
**Pull request by AnudeepYakkala**
Added a function that disables the submit button on the contact form unless there is something in all three of the fields. 

Added a function that handles a POST request to /contact. This function retrieves the data from the contact form and stores it in Datastore. Redirects back to the homepage after storing data in Datastore.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/contact")
+public class ContactServlet extends HttpServlet {
+  private static final String NAME_VALUE = "name";
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java:15]: Nit: sort these values alphabetically

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    response.sendRedirect("index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client.
+   */
+  private Optional<String> getParameter(HttpServletRequest request, String param) {
+    String value = request.getParameter(param);
+    return Optional.ofNullable(value);
```
Comment (luli0521) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java:49]: nit: inline value

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -122,6 +122,16 @@ function createListElement(text) {
   return liElement;
 }
 
+function updateContactButton() {  // eslint-disable-line no-unused-vars
+  if (document.getElementById('name').value === '' ||
+      document.getElementById('message').value === '' ||
+      document.getElementById('email').value === '') {
```
Comment (luli0521) [inline portfolio/src/main/webapp/script.js:128]: Should we consider the cases where values contain only spaces or tabs?

Comment (AnudeepYakkala) [reply portfolio/src/main/webapp/script.js]: Yea, that would be a good idea. I will add that.

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java (hunk 2 above)
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java]: sure

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java (hunk 1 above)
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java]: Okay

**Pull request merged**